src/uniolunisaar/adam/logic/synthesis/solver/symbolic/bddapproach/distrenv/safety/DistrEnvBDDGlobalSafetySolver.java

```diff
@@ -183,41 +190,58 @@
     protected void createVariables() {
         int numberOfPartitions = this.getSolvingObject().getMaxTokenCountInt();
         PLACES = new BDDDomain[2][numberOfPartitions];
+        PLACE_MULTIPLICITY = new BDDDomain[2][numberOfPartitions];
         TRANSITIONS = new BDDDomain[2];
         TOP = new BDDDomain[2];
-        RESPONSIBILITY = new BDDDomain[2];
+        RESPONSIBILITY_MULTIPLICITY = new BDDDomain[2][numberOfPartitions];
+
         for (int pos : List.of(PREDECESSOR, SUCCESSOR)) {
             for (int partition = 0; partition < numberOfPartitions; partition++) {
                 PLACES[pos][partition] = this.getFactory().extDomain(
                         this.getSolvingObject().getDevidedPlaces()[partition].size()
                                 + (this.getSolvingObject().isConcurrencyPreserving() ? 0 : 1));
-
-                /* for every system transition the system player must choose whether or not to allow that transition. */
-                TRANSITIONS[pos] = this.getFactory().extDomain(BigInteger.TWO.pow(this.getSolvingObject().getSystemTransitions().size()));
-                TOP[pos] = this.getFactory().extDomain(2);
-                RESPONSIBILITY[pos] = this.getFactory().extDomain(BigInteger.TWO.pow(numberOfPartitions));
+                /*
+                 * TODO don't create these variables if the game is safe,
+                 *  because then the multiplicity is 0 iff the partition is unmarked
+                 *  and 1 iff the partition is marked.
+                 */
```

Comment:
I did not get to this.

Reply:
For that I would open an enhancement ticket to not loose this information,